storio-content-resolver/src/main/java/com/pushtorefresh/storio/contentresolver/impl/RxChangesObserver.java

```diff
@@ -27,37 +28,67 @@ private RxChangesObserver() {
     }
 
     @NonNull
-    static Observable<Changes> observeChanges(@NonNull final ContentResolver contentResolver, @NonNull final Set<Uri> uris, @NonNull final Handler handler) {
+    static Observable<Changes> observeChanges(@NonNull final ContentResolver contentResolver,
+                                              @NonNull final Set<Uri> uris,
+                                              @NonNull final Handler handler,
+                                              final int sdkVersion) {
         return Observable.create(new Observable.OnSubscribe<Changes>() {
             @Override
             public void call(final Subscriber<? super Changes> subscriber) {
-                final ContentObserver contentObserver = new ContentObserver(handler) {
-                    @Override
-                    public boolean deliverSelfNotifications() {
-                        return false;
-                    }
+                // Use one ContentObserver for all passed Uris on API >= 16
+                if (sdkVersion >= Build.VERSION_CODES.JELLY_BEAN) {
+                    final ContentObserver contentObserver = new ContentObserver(handler) {
+                        @Override
+                        public boolean deliverSelfNotifications() {
+                            return false;
+                        }
+
+                        @Override
+                        public void onChange(boolean selfChange, Uri uri) {
+                            subscriber.onNext(Changes.newInstance(uri));
+                        }
+                    };
 
-                    @Override
-                    public void onChange(boolean selfChange, Uri uri) {
-                        subscriber.onNext(Changes.newInstance(uri));
+                    for (Uri uri : uris) {
+                        contentResolver.registerContentObserver(
```

Comment:
Jesus, mr formatting here! I guess we'll just apply codestyle to whole project at some point :)